camayoc/api.py

```diff
@@ -121,30 +126,70 @@ def __init__(self, response_handler=None, url=None):
         else:
             self.response_handler = response_handler
 
+        if self.authenticate:
+            self.login()
+
+    def login(self):
+        """Login to the server to receive an authorization token."""
+        self.authenticate = True
+        cfg = config.get_config().get('qcs', {})
+        server_username = cfg.get('username', 'admin')
+        server_password = cfg.get('password', 'pass')
+        login_request = self.request(
+            'POST',
+            urljoin(self.url, QCS_LOGIN_PATH),
+            json={
+                'username': server_username,
+                'password': server_password
+            }
+        )
+        self.token = login_request.json()['token']
+        return login_request
+
+    def logout(self):
+        """Start sending unauthorized requests.
+
+        There is no API interaction that need occur to logout.
+        We simply must send unauthorized requests.
+        """
+        self.authenticate = False
+        self.token = ''
+
+    def auth_header(self):
```

Comment:
I would call this default_headers or something like that. Because if for some reason we need to add another header in the future we would have to change all the other methods too.